src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpSystemProxy.cs

```diff
@@ -60,26 +66,86 @@ private HttpSystemProxy(WinInetProxyHelper proxyHelper, SafeWinHttpHandle sessio
 
                 if (!string.IsNullOrWhiteSpace(proxyHelper.ProxyBypass))
                 {
+                    int idx = 0;
+                    int start = 0;
+                    string tmp;
+
                     // Process bypass list for manual setting.
-                    string[] list = proxyHelper.ProxyBypass.Split(';');
-                    List<string> tmpList = new List<string>();
+                    _bypass = new List<Regex>(proxyHelper.ProxyBypass.Length / 5); // Best guess based on string length.
 
-                    foreach (string value in list)
+                    while (idx < proxyHelper.ProxyBypass.Length)
```

Comment:
I see netfx has an implementation for parsing the proxy bypass list.  Is that what this is based on?
https://referencesource.microsoft.com/#System/net/System/Net/_WebProxyDataBuilder.cs,236

Reply:
essentially yes.